### ISSUE TITLE
[Suggestion] Baritone scripting

### ISSUE THREAD

**Issue by NoNameLmao** [MeteorDevelopment/meteor-client#1350]
**Describe the feature**
It would work like any other script, but for baritone. Start with a command, then when a goal is reached/complete, automatically move on to another command.

**Example**
I know, this was meant to be "Screenshot" but I don't know how to demonstrate it via any screenshot, GIF, video. Have some text

(Current goal: None)
> script start test1
> mine 6 diamond_ore (script)
(mining)
[Baritone] Have 6 valid items
> mine 9 gold_ore (script)
[Baritone] Have 9 valid items

**My thoughts on implementation**
I think it could be done through giving baritone commands then monitoring the completion of the goal, once it was done - move to the other command in script.
I also think having a script folder in .minecraft, with Meteor reading scripts from .txt files inside and just doing commands would be cool and useful.

**Comment by MunchyG** [comment]
this would be epic for
> sel ca
> sel shift all down 4 (or expand)
> sel ca

**Comment by C10udburst** [comment]
Although for now it's fairly bare-bones, you can use [meteor-python-addon](https://github.com/C10udburst/meteor-python-addon) to make scripts that interact with baritone.


### Implementation of your example:
```py
from mcapi import baritone, event_listener
from meteordevelopment.meteorclient.events.entity.player import PickItemsEvent
from net.minecraft.item import Items

diamond_count = 0
gold_count = 0

def on_pickup(event):
    # Increase the appropriate counter
    item_type = event.itemStack.getItem()
    if item_type.equals(Items.DIAMOND):
        diamond_count += event.count
    elif item.type.equals(Items.RAW_GOLD):
        gold_count += event.count

    
    if diamond_count >= 6:
        baritone("mine gold_ore")
    if gold_count >= 9:
        baritone("stop")

# Make on_pickup run everytime an item is picked up
event_listener(on_pickup, PickupItemsEvent)

baritone("mine diamond_ore")
```

It is a bit bigger than your pseudocode, but it's the only feasible solution, as baritone doesn't (as far as I know) count the number of picked items.

**Comment by lnee94** [comment]
yes using python does work i think human wont's more of a shell scripting langueg  where it would type in stuff  to a modded
chat (so to post stuff to public chat you would haft to do chat "hello" or something) if you would have a look at the syntax of bash
you would get a better idea.
